### DISCUSSION TITLE
Configure `ctrl-r` reverse search to use entire `~/.bash_history`?

### DISCUSSION THREAD

**Discussion by jakobkmar** [junegunn/fzf#3108] [Q&A]
How can I configure fzf to search through the entire history when executing the reverse search?

Currently it only searches through a part of my history.

I don't know what the criteria for filtering the history are, but it seems like after a certain amount of time the commands are too old and won't be found by fzf anymore?
*e.g.:* `ctrl-r` with fzf only searches through 500 entries, but `sort -u ~/.bash_history | wc -l` gives 2338

System: Fedora 37 (with fzf version 0.35.1)

**Comment by sweetbbak** [comment]
https://github.com/Aloxaf/fzf-tab
https://github.com/joshskidmore/zsh-fzf-history-search

**Comment by junegunn** [comment]
It's more of a bash configuration problem. Seems related:
https://superuser.com/questions/575479/bash-history-truncated-to-500-lines-on-each-login

**Reply by lamyergeier** [reply]
@junegunn  I have following in my bash history

	shopt -s histappend                          # Append rather than overwrite history on exit; Refer: [Can history files be unified in bash? - Super User](https://superuser.com/questions/37576/can-history-files-be-unified-in-bash)
	set +o histexpand                            # Refer: [How do I escape an exclamation mark? - Super User](https://superuser.com/questions/133780/in-bash-how-do-i-escape-an-exclamation-mark)
	HISTCONTROL=ignorespace:ignoredups:erasedups # don't put duplicate lines or lines starting with space in the history.
	
	# Save 10,000 lines of history in memory
	HISTSIZE=10000                               # HISTSIZE=1000			# Default
	
	# Save 1,000,000 lines of history to disk (will have to grep ~/.bash_history for full listing)
	HISTFILESIZE=1000000                         # HISTFILESIZE=2000		
	
	# timestamp format when history command is executed in bash
	HISTTIMEFORMAT='%F %T '                      #'%Y.%m.%d %H:%M:%S' #Display timestamp (ISO 8601 standard date and a 24 hour time), always appends the unix timestamp in ~/.bash_history

	PROMPT_COMMAND="history -a; history -c; history -r; $PROMPT_COMMAND"

## My Observation

fzf search is limited to last 10000 commands (as I have defined `HISTSIZE=10000` in `~/.bashrc`)

## Question

1. So is there any issue in above? 
2. Can you please confirm the default fzf behavior: if by pressing Ctrl+r on bash, fzf uses only data from `HISTSIZE` or it uses entire `~/.bash_history`?
3. How to configure fzf to use `~/.bash_history` instead of increasing the `HISTSIZE` bash environment variable? (note: increasing `HISTSIZE` slows down bash)
